llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -6817,18 +6941,96 @@ SDValue RISCVTargetLowering::LowerOperation(SDValue Op,
     }
     return Vec;
   }
-  case ISD::LOAD:
+  case ISD::LOAD: {
+    auto *Load = cast<LoadSDNode>(Op);
+    EVT VecTy = Load->getMemoryVT();
+    // Handle normal vector tuple load.
+    if (VecTy.isRISCVVectorTuple()) {
+      MachineFunction &MF = DAG.getMachineFunction();
+      MachineFrameInfo &MFI = MF.getFrameInfo();
+      SDLoc DL(Op);
+      MVT XLenVT = Subtarget.getXLenVT();
+      unsigned NF = VecTy.getVectorMinNumElements();
+      unsigned Sz = VecTy.getSizeInBits();
+      unsigned NumElts = Sz / (NF * 8);
+      int Log2LMUL = Log2_64(NumElts) - 3;
+
+      auto Flag = SDNodeFlags();
+      Flag.setNoUnsignedWrap(true);
+      SDValue Ret = DAG.getUNDEF(VecTy);
+      SDValue BasePtr = Load->getBasePtr();
+      if (auto *FI = dyn_cast<FrameIndexSDNode>(BasePtr))
+        MFI.setStackID(FI->getIndex(), TargetStackID::ScalableVector);
+      SDValue VROffset = DAG.getNode(RISCVISD::READ_VLENB, DL, XLenVT);
+      VROffset =
+          DAG.getNode(ISD::SHL, DL, XLenVT, VROffset,
+                      DAG.getConstant(std::max(Log2LMUL, 0), DL, XLenVT));
+
+      // Load NF vector registers and combine them to a vector tuple.
+      for (unsigned i = 0; i < NF; ++i) {
+        SDValue LoadVal = DAG.getLoad(
+            MVT::getScalableVectorVT(MVT::i8, NumElts), DL, DAG.getEntryNode(),
+            BasePtr, Load->getPointerInfo(), Align(8));
```

Comment:
Won't this PointerInfo make it look like all parts of the tuple start at the same address?

Reply:
I think it won't since the `BasePtr` is increased by `VLENB` every iteration, there's a test for tuple load/store in `alloca-load-store-vector-tuple.ll`.

Reply:
MachinePointerInfo contains a Value* and an integer offset. This is used by alias analysis to reference back to the IR. By copying the MachinePointerInfo to every load/store without changing the offset, you're telling alias analysis that they all start at the same address.

I don't know if it's possible to pass a scalable value to the offset in MachinePointerInfo yet. You might have to create a new MachinePointerInfo that does have a Value* and only contains the address space.

Reply:
What the `Value*` should be in this case?

Reply:
Sorry I made a typo. It should have been "does not have a Value*"

Reply:
Got it!